app/policies/application_policy.rb

```diff
@@ -26,7 +26,7 @@ def new?
   end
 
   def update?
-    user.present? && (record.user == user || user.role == admin)
+    user.present? && (record.public == nil || user.admin? || record.user == user)
```

Comment:
Consider writing a `WikiPolicy` rather than customizing the parent policy that all policies inherit from. `record.public` wouldn't be relevant to other policies that inherit from `ApplicationPolicy` and belongs in a `WikiPolicy`.

`record.public == nil` can be simplified to `!record.public`

It looks like `update?` will always return `true` for a private wiki, meaning any logged in user can update a private wiki.

Think about writing a policy spec to check that the policy is doing what you need it to. Here's how to write a policy spec: https://github.com/elabs/pundit#rspec